tabular/src/autogluon/tabular/models/abstract/abstract_model.py

```diff
@@ -707,3 +701,73 @@ def save_info(self) -> dict:
         json_path = self.path + self.model_info_json_name
         save_json.save(path=json_path, obj=info)
         return info
+
+class AbstractNeuralNetworkModel(AbstractModel):
+    # TODO: v0.1 clean method
+    def _get_types_of_features(self, df, skew_threshold=None, embed_min_categories=None, use_ngram_features=None, needs_torch=False, needs_extra_types=True):
+        """ Returns dict with keys: : 'continuous', 'skewed', 'onehot', 'embed', 'language', values = ordered list of feature-names falling into each category.
+            Each value is a list of feature-names corresponding to columns in original dataframe.
+            TODO: ensure features with zero variance have already been removed before this function is called.
+        """
+        if self.types_of_features is not None:
```

Comment:
if self.types_of_features exists in this class but not to abstractmodel, add it to the `__init__` call of this class as a variable declaration. Then it can be removed from the inheriting models `__init__` call.

Reply:
Added self._types_of_features to this __init__ and removed it from TabTransformerModel's __init__